src/graphics/color.js

```diff
@@ -59,17 +59,18 @@ Crafty.extend({
         }
 
         function parseHexString(hex, c) {
-            var l;
-            if (hex.length === 7){
-                l=2;
-            } else if (hex.length === 4){
-                l=1;
-            } else {
+            if (hex.length === 4){
+                var r = hex.substr(1, 1);
+                var g = hex.substr(2, 1);
+                var b = hex.substr(3, 1);
+                hex = '#'+r+r+g+g+b+b;
+            }
+            if (hex.length !== 7) {
                 return default_value(c);
             }
-            c._red = parseInt(hex.substr(1, l), 16);
-            c._green = parseInt(hex.substr(1+l, l), 16);
-            c._blue = parseInt(hex.substr(1+2*l, l), 16);
+            c._red = parseInt(hex.substr(1, 2), 16);
+            c._green = parseInt(hex.substr(3, 2), 16);
+            c._blue = parseInt(hex.substr(5, 2), 16);
             return c;
         }
```

Comment:
Ok I'll update it and I'll squash merge it back into develop
Op wo 4 mei 2016 om 14:18 schreef mucaho <notifications@github.com>

> In src/graphics/color.js
> <https://github.com/craftyjs/Crafty/pull/1033#discussion_r62028824>:
>
> >                  return default_value(c);
> >              }
> > -            c._red = parseInt(hex.substr(1, l), 16);
> > -            c._green = parseInt(hex.substr(1+l, l), 16);
> > -            c._blue = parseInt(hex.substr(1+2*l, l), 16);
> > +            c._red = parseInt(hex.substr(1, 2), 16);
> > +            c._green = parseInt(hex.substr(3, 2), 16);
> > +            c._blue = parseInt(hex.substr(5, 2), 16);
> >              return c;
> >          }
>
> You could probably optimize this code slightly by removing redundant calls
> to substr:
>
> function parseHexString(hex, c) {
>     var r, g, b,
>         l = hex.length;
>
>     if (l === 7) {
>         r = hex.substr(1, 2);
>         g = hex.substr(3, 2);
>         b = hex.substr(5, 2);
>     } else if (l === 4) {
>         r = hex.substr(1, 1); r += r;
>         g = hex.substr(2, 1); g += g;
>         b = hex.substr(3, 1); b += b;
>     } else {
>         return default_value(c);
>     }
>     c._red = parseInt(r, 16);
>     c._green = parseInt(g, 16);
>     c._blue = parseInt(b, 16);
>
>     return c;
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/craftyjs/Crafty/pull/1033/files/c3f01a5b45e8e372615effddc1205f2901e87371#r62028824>
>